### PR TITLE
create indexes in mgodatagen mode

### PR DESCRIPTION
**Pull request by SergeAx**
I was unable to pass all tests because a) I run mongo in docker locally, so don't have `mongo` command b) two tests just failed silently, no idea why, perhaps of my environment and lack of some other utilities. Let's see what Github CI have to say.

I fixed some typos, it's about some guy called Asperger, if you love those typos and want them to stay - changes in a seperate commit and I can just rebase without it.

### CODE REVIEW

**Comment by SergeAx** [comment]
My local test data, for further investigation:

```JSON
[
  {
    "collection": "collection",
    "count": 20,
    "content": {
      "array1": {
        "type": "array",
        "size": 5,
        "arrayContent": {
          "type": "int",
          "minInt": 0,
          "maxInt": 10
        }
      },
      "array2": {
        "type": "array",
        "size": 5,
        "arrayContent": {
          "type": "int",
          "minInt": 0,
          "maxInt": 10
        }
      }
    },
    "indexes": [
      {
        "name": "array1",
        "key": {
          "array1": 1
        }
      },
      {
        "name": "array2",
        "key": {
          "array2": 1
        }
      }
    ]
  }
]
```

Test request:
```JSON
db.collection.find({
  "array1": 1,
  "array2": 1
})
```

**Review by feliixx — CHANGES_REQUESTED** [review]
Hi @SergeAx, thanks a lot for the PR ! 

Overall looks great :) can you please rebase it on master, I've updated the test workflow to run on both `push` and `pull_request`

Also, this needs a test. Can you please add one in `run_test.go` ?

**Comment by SergeAx** [comment]
Pulled your master, CI is working now. Off to writing a test.

**Comment by SergeAx** [comment]
Here, check out the test. `codecov/patch` shows only `return ..., err` strings as red, so the happy path is fully covered.

File: run.go
Comment (feliixx) [inline run.go:441]: Do you want to be able to run this type of query from front-end, or is this just for testing purpose ?

**Comment by feliixx** [comment]
thanks for adding the test ! I've left only a small comment, no worries for code coverage

File: run.go
Comment (SergeAx) [reply run.go:441]: I made it as a quick workaround for testing and don't think it has to be available for end-user. The final decision is up to you as a maintainer.

Comment (feliixx) [reply run.go:441]: ok, fine for me. Allowing new method in the front-end requires a few changes, and I don't think it's worth the trouble

**Pull request merged**
